content/src/components.ts

```diff
@@ -299,33 +295,45 @@ export async function initComponentsWithEnv(env: Environment): Promise<AppCompon
     clock
   })
 
-  const controller = new Controller(
-    {
-      env,
-      challengeSupervisor,
-      deployer,
-      logs,
-      metrics,
-      database,
-      sequentialExecutor,
-      activeEntities,
-      denylist,
-      fs,
-      snapshotGenerator,
-      failedDeployments,
-      contentCluster,
-      synchronizationState,
-      storage
-    },
-    ethNetwork
-  )
-
   const migrationManager = MigrationManagerFactory.create({ logs, env })
 
   env.logConfigValues(logs.getLogger('Environment'))
 
-  const server = new Server({ controller, metrics, env, logs, fs })
+  const _server = await createServerComponent<GlobalContext>(
```

Comment:
Is this wrapper needed? I think it WKC server already handles those cases

Reply:
yes, super annoying, I'll send a PR to WKC http-server in the future, currently it fails with an error if you stop a non-started server, which breaks our tests